upgrade.go

```diff
@@ -0,0 +1,231 @@
+package upgrade
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/pkg/errors"
+	"github.com/termie/go-shutil"
+)
+
+func CurrentVersion(version string) (semver.Version, error) {
+	return semver.Make(version)
+}
+
+func AvailableVersion(repo string, filePrefix string, channel string) (semver.Version, error) {
```

Comment:
I got an idea what the tool is and understood a bit the functionality used. Just for my knowledge where are all the variables called defined. I see the version and architecture. but i dont see the repo.